### PR TITLE
modified alignment geometry templates

### PR DESCRIPTION
**Pull request by SergejMuhic**
1. [adding Curve2D representation to Alignment Geometry and AG Gradient](https://github.com/bSI-InfraRoom/IFC-Specification/commit/8e2f2af96a6fa38cec5188561a1f6fa01bc6ecc4)

Adding a 2D representation of alignment since measurements along 2D and 3D are required.

Discussion:
Add all valid alignment representations i. e. IfcPolyline, IfcOffsetCurveByDistances etc. to the Curve2D and Curve3D representations?

### CODE REVIEW

**Review by larswik — COMMENTED** [review]
I think that the documerntation for Alignment Geometry needs amendment (regarding RepresentationType="Curve2D") and documentation for Alignment Geometry Gradient needs to be added to clarify how this works.

File: IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/PropertySets/Pset_TransportationDeviceCommon/Documentation.md [hunk 1]
```diff
@@ -1 +1 @@
-Properties common to the definition of all occurrences of IfcTransportationDevice or IfcTransportationDeviceType
\ No newline at end of file
+Properties common to the definition of all occurrences of IfcTransportationDevice or IfcTransportationDeviceType
```
Comment (pjanck) [inline IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/PropertySets/Pset_TransportationDeviceCommon/Documentation.md]:
```suggestion
Properties common to the definition of all occurrences of _IfcTransportationDevice_ or _IfcTransportationDeviceType_
```

File: IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/DocTemplateDefinition.xml [hunk 2]
```diff
@@ -50,6 +50,63 @@
...
+														</DocModelRuleConstraint>
+														<DocModelRuleConstraint>
+															<Expression xsi:type="DocOpStatement" Operation="compareequal">
+																<Reference xsi:type="DocOpReference">
+																	<EntityRule xsi:type="DocModelRuleEntity" xsi:nil="true" href="IfcLabel_1UYMuwZ1566P968r8_v2Pz" />
+																</Reference>
+																<Value xsi:type="DocOpLiteral" Operation="loadstring" Literal="Axis" />
+															</Expression>
+														</DocModelRuleConstraint>
+													</Rules>
```
Comment (pjanck) [inline IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/DocTemplateDefinition.xml]: Added twice - serialization problem perhaps?

File: IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/DocTemplateDefinition.xml [hunk 3]
```diff
@@ -50,6 +50,63 @@
...
+														</DocModelRuleConstraint>
+														<DocModelRuleConstraint>
+															<Expression xsi:type="DocOpStatement" Operation="compareequal">
+																<Reference xsi:type="DocOpReference">
+																	<EntityRule xsi:type="DocModelRuleEntity" xsi:nil="true" href="IfcLabel_0hxnaRHwbFuxyZ3b2iuiyK" />
+																</Reference>
+																<Value xsi:type="DocOpLiteral" Operation="loadstring" Literal="Curve2D" />
+															</Expression>
+														</DocModelRuleConstraint>
+													</Rules>
```
Comment (pjanck) [inline IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/DocTemplateDefinition.xml]: Simile.

File: IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Alignment Geometry Gradient/DocTemplateDefinition.xml [hunk 4]
```diff
@@ -37,6 +37,51 @@
...
+										<DocModelRuleAttribute Name="RepresentationIdentifier">
+											<Rules>
+												<DocModelRuleEntity xsi:nil="true" href="IfcLabel_1UYMuwZ1566P968r8_v2Pz" />
+											</Rules>
+										</DocModelRuleAttribute>
+										<DocModelRuleAttribute Name="RepresentationType">
+											<Rules>
+												<DocModelRuleEntity xsi:nil="true" href="IfcLabel_0hxnaRHwbFuxyZ3b2iuiyK" />
+											</Rules>
+										</DocModelRuleAttribute>
```
Comment (pjanck) [inline IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Alignment Geometry Gradient/DocTemplateDefinition.xml:56]: Literals for comparison missing.

**Review by pjanck — CHANGES_REQUESTED** [review]

File: IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Alignment Geometry Gradient/DocTemplateDefinition.xml (hunk 4 above)
Comment (SergejMuhic) [reply IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Alignment Geometry Gradient/DocTemplateDefinition.xml:56]: Think this is also resolved with the latest commit. Please check and resolve the comment.

Comment (pjanck) [reply IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Alignment Geometry Gradient/DocTemplateDefinition.xml:56]: This doesn't seem to be resolved. Or are there no `RepresentationIdentifier` and `RepresentationType` constraints imposed by this concept template?

**Review by pjanck — CHANGES_REQUESTED** [review]

File: IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Alignment Geometry Gradient/DocTemplateDefinition.xml (hunk 4 above)
Comment (SergejMuhic) [reply IFC4x3/Templates/Product Shape/Product Geometric Representation/Alignment Geometry/Alignment Geometry Gradient/DocTemplateDefinition.xml:56]: Are you expecting the value constraints on sub templates? Those are not necessary. See `Alignment Geometry Cant` for instance.

**Review by larswik — APPROVED** [review]

**Review by pjanck — APPROVED** [review]

**Pull request merged**